packages/richtext-lexical/src/features/blocks/client/plugin/index.tsx

```diff
@@ -56,22 +56,14 @@ export const BlocksPlugin: PluginComponent = () => {
 
             if ($isRangeSelection(selection)) {
               const blockNode = $createBlockNode(payload)
-              // Insert blocks node BEFORE potentially removing focusNode, as $insertNodeToNearestRoot errors if the focusNode doesn't exist
-              $insertNodeToNearestRoot(blockNode)
 
               const { focus } = selection
               const focusNode = focus.getNode()
+              // Insert blocks node BEFORE potentially removing focusNode, as $insertNodeToNearestRoot errors if the focusNode doesn't exist
+              $insertNodeToNearestRoot(blockNode)
```

Comment:
This is the key point of the fix. We still insert before removing focusNode, but after getting focusNode, since `$insertNodeToNearestRoot` can change the selection.